Source/Diagnostics/FullDiagnostics.cpp

```diff
@@ -422,21 +422,8 @@ FullDiagnostics::InitializeFieldFunctors (int lev)
         } else if ( m_varnames[comp] == "jz" ){
             m_all_field_functors[lev][comp] = std::make_unique<CellCenterFunctor>(warpx.get_pointer_current_fp(lev, 2), lev, m_crse_ratio);
         } else if ( m_varnames[comp] == "rho" ){
-            if ( WarpX::do_back_transformed_diagnostics ) {
-                if ( WarpX::maxwell_solver_id == MaxwellSolverAlgo::PSATD ) {
-                    // rho_new is stored in component 1 of rho_fp when using PSATD
-                    amrex::MultiFab *rho_new = new amrex::MultiFab(*warpx.get_pointer_rho_fp(lev), amrex::make_alias,
-                                                                   1, 1);
-                    m_all_field_functors[lev][comp] = std::make_unique<CellCenterFunctor>(rho_new, lev, m_crse_ratio);
-                } else {
-                    m_all_field_functors[lev][comp] = std::make_unique<CellCenterFunctor>(warpx.get_pointer_rho_fp(lev),
-                                                                                          lev, m_crse_ratio);
-                }
-            }
-            else {
-                // Initialize rho functor to dump total rho
-                m_all_field_functors[lev][comp] = std::make_unique<RhoFunctor>(lev, m_crse_ratio);
-            }
+            // Initialize rho functor to dump total rho
+            m_all_field_functors[lev][comp] = std::make_unique<RhoFunctor>(lev, m_crse_ratio);
```

Comment:
@RevathiJambunathan Do you agree that this is now correct?

Reply:
Yes @EZoni Looks good to me!